doc/dev/kclient_dev_bootstrap.rst

```diff
@@ -0,0 +1,474 @@
+The VMs will share the host’s DNS resolution, enabling seamless name resolution
+for addresses within the Sepia network.
+
+Build the Kernel
+----------------
+
+.. note:: The text assumes all prerequists for building the kernel are
+   installed both on the host and guest.
+
+Clone the Git Repository
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start by cloning the kernel source repository into the shared folder so that it
+is accessible from both the host and guest.
+
+.. code-block:: bash
+
+    git init linux && cd linux
+    git remote add torvalds git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+    git remote add ceph https://github.com/ceph/ceph-client.git
+    git fetch && git checkout torvalds/master
+
+Configure the Kernel
+^^^^^^^^^^^^^^^^^^^^
+
+From within the **guest VM**, copy the current kernel configuration file from
+``/boot/config-$(uname -r)`` to ``.config`` in the kernel source directory.
+This ensures that the new build will have a configuration similar to the
+currently running guest kernel.
+
+.. code-block:: bash
+
+   cp /boot/config-$(uname -r) /path/to/linux/.config
+   make localmodconfig
+   make olddefconfig
+
+- ``make localmodconfig``: Updates ``.config`` to include only the modules
+  currently loaded on the host.
+- ``make olddefconfig``: Fills in any missing configuration options in
+  ``.config`` with default values.
+
+For more information on this process, refer to `the localmodconfig
+documentation at
+<https://www.kernel.org/doc/Documentation/admin-guide/README.rst>`_. This
+command will ensure that all necessary modules are configured.
+
+.. note:: You can alternatively use the `Ceph Kernel QA Config`_ for building
+   the kernel.
+
+We now have a kernel config with reasonable defaults for the architecture
+you're building on. The next thing to do is to enable configs which will build
+Ceph and provide the functionality needed to do testing.
+
+.. code-block:: bash
+
+    cat > ~/.ceph.config <<EOF
+    CONFIG_CEPH_LIB=m
+    CONFIG_CEPH_FS=m
+    CONFIG_CEPH_FSCACHE=y
+    CONFIG_CEPH_FS_POSIX_ACL=y
+    CONFIG_CEPH_FS_SECURITY_LABEL=y
+    CONFIG_CEPH_LIB_PRETTYDEBUG=y
+    CONFIG_DYNAMIC_DEBUG=y
+    CONFIG_DYNAMIC_DEBUG_CORE=y
+    CONFIG_FRAME_POINTER=y
+    CONFIG_FSCACHE=y
+    CONFIG_FSCACHE_STATS=y
+    CONFIG_FS_ENCRYPTION=y
+    CONFIG_FS_ENCRYPTION_ALGS=y
+    CONFIG_KGDB=y
+    CONFIG_KGDB_SERIAL_CONSOLE=y
+    CONFIG_XFS_FS=y
+    EOF
+
+In addition to enabling Ceph-related configs, we are also enabling some useful
+debug configs and XFS (as an alternative to ext4 if needed for our root file
+system).
+
+Merge the configs:
+
+
+.. code-block:: bash 
+
+
+   scripts/kconfig/merge_config.sh .config ~/.ceph.config
+
+
+Build the Kernel
+^^^^^^^^^^^^^^^^
+
+Compile the kernel using one of the following commands. Adjust the ``-j`` value
+as needed to match the number of cores available on the host for parallel
+processing.
+
```

Comment:
For your workflow, would this be done on the host or the guest VM? (I prefer doing this on the host; I don't really understand building the kernel within the VM. I understand it makes installing the modules easier but I don't really see that as necessary to begin with.)

Reply:
You build on the host, you install on the guest